### PR TITLE
OPS-623 - Build pipeline

### PR DESCRIPTION
**Pull request by mamutmk5**
# Description

Add a automatic build pipeline and change the test steps.
Automatic build pipeline is trigert by push to branches, pullrequest and ignore some paths.
Automatic test stepst is know trigert by push to branches, pullrequest and ignore some paths.
Triger banches:
- master
- develop
- release/*
- hotfix/*
Ignore paths:
- /.github/**

## Links to Tickets or other pull requests
- https://ticketsystem.schul-cloud.org/browse/SC-4616

## Changes
Build and test pipeline

## Deployment
No Deployment, change pipelines at github

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: .github/workflows/pr-automation-auto-assign.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: PR Automation
```
Comment (reneoelke) [inline .github/workflows/pr-automation-auto-assign.yml:1]: This change is out of context.

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+name: Node.js CI
```
Comment (reneoelke) [inline .github/workflows/nodejs.yml:1]: This change is out of context.

File: .github/workflows/build-and-push-docker-image.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: Build and push Docker image
```
Comment (reneoelke) [inline .github/workflows/build-and-push-docker-image.yaml:1]: Sync this file with https://github.com/schul-cloud/schulcloud-synapse-synchronization/blob/develop/.github/workflows/build-and-push-docker-image.yaml

Comment (mamutmk5) [reply .github/workflows/build-and-push-docker-image.yaml:1]: Change to the version from schulcloud-synapse-synchronization.

File: .github/workflows/nodejs.yml (hunk 2 above)
Comment (mamutmk5) [reply .github/workflows/nodejs.yml]: File removed
https://github.com/schul-cloud/node-notification-service/pull/205/commits/ed12ab488e29e057410f4a904dccf68a71e107c9

File: .github/workflows/pr-automation-auto-assign.yml (hunk 1 above)
Comment (mamutmk5) [reply .github/workflows/pr-automation-auto-assign.yml]: File Removed
https://github.com/schul-cloud/node-notification-service/pull/205/commits/b0dfe1eba549fded8b9ebd7fe7d6e39bb5798f15

**Pull request merged**